test/unit/event_machine_proxy_test.rb

```diff
@@ -89,45 +106,44 @@
     end
 
     it "add_timer" do
-      mock(ExceptionHandling).log_error(EXCEPTION, "add_timer")
+      mock(ExceptionHandling).log_error(EXCEPTION, "add_timer", {})
       mock(EventMachine::Synchrony).add_timer(10) { |duration, *args| args.first.call }
       @em.add_timer(10) { raise EXCEPTION }
     end
 
     it "add_periodic_timer" do
-      mock(ExceptionHandling).log_error(EXCEPTION, "add_periodic_timer")
+      mock(ExceptionHandling).log_error(EXCEPTION, "add_periodic_timer", {})
       mock(EventMachine::Synchrony).add_periodic_timer(10) { |duration, *args| args.first.call }
       @em.add_periodic_timer(10) { raise EXCEPTION }
     end
 
     it "next_tick" do
-      mock(ExceptionHandling).log_error(EXCEPTION, "next_tick")
+      mock(ExceptionHandling).log_error(EXCEPTION, "next_tick", {})
       mock(EventMachine::Synchrony).next_tick { |*args| args.first.call }
       @em.next_tick { raise EXCEPTION }
     end
   end
 
-  [false, true].each do |synchrony|
-    describe "synchrony = #{synchrony}" do
-      it "should dispatch to the proxy's synchrony method instead of run iff synchrony" do
-        proxy_mock = Struct.new(:proxy, :class_connection) do
-          if synchrony
-            def self.synchrony(&block)
-              block.(:synchrony)
-            end
-          end
-
-          def self.run(&block)
-            block.(:run)
-          end
+  describe "run" do
+    { synchrony: SynchronyProxyMock, run: RunProxyMock }.each do |method, proxy_mock|
+      describe "using #{method}" do
+        before do
+          @proxy = ExceptionalSynchrony::EventMachineProxy.new(proxy_mock, nil)
         end
 
-        mock(proxy_mock).error_handler
-
-        proxy = ExceptionalSynchrony::EventMachineProxy.new(proxy_mock, nil)
+        it "should dispatch to the proxy's synchrony method instead of run iff synchrony" do
+          dispatched = false
+          block = -> { dispatched = true }
```

Comment:
Since the named block is used just once, how about simplifying to inline?
```
assert_equal method, @proxy.run { dispatched = true }
```
Same comment below.